gnosis/eth/account_abstraction/user_operation.py

```diff
@@ -44,27 +44,65 @@ def from_bundler_response(
-            HexBytes(user_operation_response["userOperation"]["initCode"]),
-            HexBytes(user_operation_response["userOperation"]["callData"]),
-            int(user_operation_response["userOperation"]["callGasLimit"], 16),
-            int(user_operation_response["userOperation"]["verificationGasLimit"], 16),
-            int(user_operation_response["userOperation"]["preVerificationGas"], 16),
-            int(user_operation_response["userOperation"]["maxFeePerGas"], 16),
-            int(user_operation_response["userOperation"]["maxPriorityFeePerGas"], 16),
-            HexBytes(user_operation_response["userOperation"]["paymasterAndData"]),
-            HexBytes(user_operation_response["userOperation"]["signature"]),
-            ChecksumAddress(user_operation_response["entryPoint"]),
-            metadata=UserOperationMetadata(
-                HexBytes(user_operation_response["transactionHash"]),
-                HexBytes(user_operation_response["blockHash"]),
-                int(user_operation_response["blockNumber"], 16),
-            ),
+    ) -> Union["UserOperation", "UserOperationV07"]:
+        user_operation = user_operation_response["userOperation"]
+        metadata = UserOperationMetadata(
+            HexBytes(user_operation_response["transactionHash"]),
+            HexBytes(user_operation_response["blockHash"]),
+            int(user_operation_response["blockNumber"], 16),
         )
+        if "initCode" in user_operation:
+            return cls(
+                HexBytes(user_operation_hash),
+                ChecksumAddress(user_operation["sender"]),
+                int(user_operation["nonce"], 16),
+                HexBytes(user_operation["initCode"]),
+                HexBytes(user_operation["callData"]),
+                int(user_operation["callGasLimit"], 16),
+                int(user_operation["verificationGasLimit"], 16),
+                int(user_operation["preVerificationGas"], 16),
+                int(user_operation["maxFeePerGas"], 16),
+                int(user_operation["maxPriorityFeePerGas"], 16),
+                HexBytes(user_operation["paymasterAndData"]),
+                HexBytes(user_operation["signature"]),
+                ChecksumAddress(user_operation_response["entryPoint"]),
+                metadata=metadata,
+            )
+        else:
+            if paymaster := user_operation.get("paymaster"):
+                # Paymaster parameters are optional
+                paymaster_verification_gas_limit = int(
+                    user_operation["paymasterVerificationGasLimit"], 16
+                )
+                paymaster_post_op_gas_limit = int(
+                    user_operation["paymasterPostOpGasLimit"], 16
+                )
+                paymaster_data = HexBytes(user_operation["paymasterData"])
+            else:
+                paymaster_verification_gas_limit = None
+                paymaster_post_op_gas_limit = None
+                paymaster_data = None
```

Comment:
This is just a matter of taste, but it is my understanding that avoiding "else" like this is clearer. But it's your decision :) 

```suggestion
           paymaster_verification_gas_limit = None
           paymaster_post_op_gas_limit = None
           paymaster_data = None
           
           if paymaster := user_operation.get("paymaster"):
                # Paymaster parameters are optional
                paymaster_verification_gas_limit = int(
                    user_operation["paymasterVerificationGasLimit"], 16
                )
                paymaster_post_op_gas_limit = int(
                    user_operation["paymasterPostOpGasLimit"], 16
                )
                paymaster_data = HexBytes(user_operation["paymasterData"])                
```

Reply:
In my opinion in this case it is more confusing. You are defining variables to replace them later if a condition matches. But I will keep that in mind, thanks for the suggestion